### PR TITLE
Switch to distribution v3

### PR DESCRIPTION
**Pull request by bacongobbler**
Replaces our current usage of docker/distribution with a more up-to-date release.

See https://github.com/oras-project/oras-go/pull/13 for a similar change being made upstream. Once upstream's change has been merged and a new release has been cut, we can drop the dependency on docker/distribution.

### CODE REVIEW

**Review by cndoit18 — APPROVED** [review]

File: go.mod
Comment (hickeyma) [inline go.mod:16]: Should this be removed?

**Review by hickeyma — COMMENTED** [review]
Should the docker/distribution still be referenced?

File: go.mod
Comment (bacongobbler) [reply go.mod:16]: I mentioned this in the OP:

> Once upstream's change has been merged and a new release has been cut, we can drop the dependency on docker/distribution.

We have to wait until oras-go has merged https://github.com/oras-project/oras-go/pull/13 and cut a new release before we can drop this dependency

Comment (hickeyma) [reply go.mod:16]: Ok, sorry missed that

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @bacongobbler

**Review by technosophos — APPROVED** [review]
LGTM 2

**Pull request merged**
